leetcode/src/28.c

```diff
@@ -0,0 +1,37 @@
+int strStr(char* haystack, char* needle) {
```

Comment:
I think your way is brutal force. Do you come up with more efficient solution? It is a typical problem for pattern search. For example: `Rabin Karp`, `KMP`, or `Horspool`, ...

Reply:
I have implemented this a year ago and I do not want to blindly implement an algorithm, so I'll need some time for that. It is up to you, you can merge this PR and I'll raise another one with the optimised solution to this problem or keep it open till I come up with another approach.

Reply:
should I raise a new pull request or add the `kmp` implementation to this pull request.

Reply:
Yes, if you have the chance

Reply:
Hi, I could not see my commits in mater branch.

Reply:
You can create a new one

Reply:
ok, I'll create a new PR qwith all the previous and will add KMP to it.